Add cmdline argument to install/use a specific tag

irkode opened the pull request:
Implements #106 

* add comandline argument to use and install
* replace --latest with pseudo tag "latest"
* supports semver abbreviations to ease up typing
* add unit test for tag determination
* add/update testscripts

improvements
- split up cmd_test.go to have separate functions so one can only run a subset
- refactored commandline flags and test to match the implementation
- enabled tests defined in cmd/install

irkode committed:
cmd/testscript: enable test execution per subcommand

irkode committed:
cmd/all: add cmdline arg to install/use a specific tag (obsoleting --latest flag)

irkode committed:
cmd/status: add missing mutual exclusive check for --print*

irkode committed:
cmd/status: fix error message if tag does not exist (test: status_err_3)

jmooring commented:
Let's keep the CLI arg simple, one of: `v1.2.3`, `1.2.3` or `latest`.

If you need a unit test for something in use.go, place it in use_test.go.

irkode commented:
simplified the commandline arg as per your comment. (`v1.2.3`, `1.2.3` or `latest`)
moved unit test to use_test.go and adapted to the simpler logic. (would keep them cause the func is called transitive by three commands.)

jmooring commented:
@irkode Many thanks for your contribution. Much appreciated. I'll make some minor changes in a separate PR, and roll a release this weekend.
